server/api/messaging.py

```diff
@@ -0,0 +1,207 @@
+from flask import request, Blueprint, jsonify
+from sqlalchemy.orm.exc import NoResultFound
+from sqlalchemy import or_, and_
+from models.message import Message, Conversation
+from models.user import User
+from db import session
+from .auth import authenticate
+
+messaging_views = Blueprint('messaging', __name__, url_prefix="chat")
+
+@messaging_views.route('/new_message', methods=["POST"])
+@authenticate
+def new_message(**kwargs):
+    response_dict = {}
+    user = kwargs['user']
+    request_dict = request.get_json()
+    receiver_id = request.get('receiver_id', None)
+    receiver = User.get_instance(**{'id': receiver_id})
+    if receiver_id == None:
+        response_dict['status'] = 400
+        response_dict['message'] = "Cannot create a new message with no receiving user"
+        return jsonify(response_dict), 400
```

Comment:
So we might want to think about this. There's a two cases for new message:
1. conversation doesn't exist
2. conversation already exists

If the conversation doesn't exist, we need both sets of user ids, then we need to make a new conversation and push the message into that conversation.

If the conversation already exists, we need to send the conversation id and the sender id, then we can just push that message into the conversation. In this situation we don't need the receiver id

Reply:
So are you saying to do a check to see if the conversation exists and if so, don't deal with the receiver and instead push the new message to the existing conversation object and send the conversation id to the front end? Or are you saying that we can get the receiver from the conversation if it exists?
I'm a bit unclear on what you mean when you say "send the conversation id and the sender id"